### PR TITLE
Use beacon state interfaces instead of concrete structs

### PR DESCRIPTION
**Pull request by rkapka**
**THIS PR IS A REPLACEMENT FOR #10182, WHICH WAS CLOSED BECAUSE OF TECHNICAL ISSUES**

**What type of PR is this?**

Cleanup

**What does this PR do? Why is it needed?**

We should use beacon state interfaces instead of structs wherever possible for better abstraction and to avoid leaking internal data. Using interfaces will also help reduce the impact of applying the native state feature flag.

### CODE REVIEW

File: beacon-chain/powchain/service.go [hunk 1]
```diff
@@ -281,8 +281,8 @@ func (s *Service) ChainStartEth1Data() *ethpb.Eth1Data {
 
 // PreGenesisState returns a state that contains
 // pre-chainstart deposits.
-func (s *Service) PreGenesisState() state.BeaconState {
-	return s.preGenesisState
+func (s *Service) PreGenesisState() (state.BeaconState, error) {
+	return s.preGenesisState, nil
```
Comment (potuz) [inline beacon-chain/powchain/service.go:285]: Why this change if we never return an error?

Comment (potuz) [inline beacon-chain/powchain/service.go:285]: Why this change if we never return an error?

Comment (rkapka) [reply beacon-chain/powchain/service.go:285]: This service implements an interface. Another implementation needs to return an error.
https://github.com/prysmaticlabs/prysm/blob/8a10f4e87e076fc50b06fc24efbf31a6dffe6c95/beacon-chain/deterministic-genesis/service.go#L135-L141

**Review by nisdas — APPROVED** [review]
